src/components/ecl-button/src/styles/ecl-button-ec.scss

```diff
@@ -1,35 +1,24 @@
-@use '@ecl/theme-ec';
-@use '@ecl/vanilla-component-icon/icon-ec';
-@use '@ecl/vanilla-component-link/link-ec';
-@use '@ecl/vanilla-component-button/button-ec';
+@use '@ecl/theme-ec/theme' as theme;
+@use '@ecl/theme-ec/variables' as var;
+@use '@ecl/button/button' with (
+  $theme: theme.$theme,
+  $button: var.$button
+);
 
 :host {
   display: contents;
 }
 
-.ecl-button__icon {
-  vertical-align: middle;
+.ecl-button--icon-only ecl-icon {
```

Comment:
You cannot style the custom element, you have to use the ::slotted pseudo element the same way you do right after this. 
![image](https://github.com/user-attachments/assets/06df5891-031b-4035-8b69-3e10c8a13732)